documentation/src/main/asciidoc/topics/ref_rest_caches_query.adoc

```diff
@@ -59,12 +59,22 @@ include::json/rest_query_request_body.json[]
 ----
 
 [id='rest_v2_query_reindex']
-== Re-indexing Data
-Re-index all data in caches with `POST` requests and the `?action=mass-index` parameter.
+== Rebuilding indexes
+When you change existing Protobuf schema definition at runtime or when the index is persistent, you must reindex the data in the index.
+Rebuilding indexes can take a long time to complete because the process takes place for all data in the grid.
+While the rebuild operation is in progress, queries might also return fewer results.
+
+[NOTE]
```

Comment:
@wfink mentioned this information. I'm not sure if it applies to reindexing and also updating indexes.

Reply:
not sure what you mean with reindex vs. updating index.
To me reindex is to delete and create a new one - because every entry changed because of config
updating index is to change one entry because the data of it changed (which is the ongoing work)